opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -203,16 +203,20 @@ def create(
 
         resource_detector: str
         for resource_detector in otel_experimental_resource_detectors:
-            resource_detectors.append(
-                next(
-                    iter(
-                        entry_points(
-                            group="opentelemetry_resource_detector",
-                            name=resource_detector.strip(),
-                        )  # type: ignore
-                    )
-                ).load()()
-            )
+            try:
+                resource_detectors.append(
+                    next(
+                        iter(
+                            entry_points(
+                                group="opentelemetry_resource_detector",
+                                name=resource_detector.strip(),
+                            )  # type: ignore
+                        )
+                    ).load()()
+                )
+            except:
+                # TODO emit warning per failed detector?
```

Comment:
The auto-instrumentation code does this when it fails to load an instrumentor:

```
            _logger.exception(
                "Importing of %s failed, skipping it", entry_point.name
            )
```

Reply:
Good suggestion, I'll add that - thanks!